src/common/import_session.c

```diff
@@ -332,15 +348,29 @@ static const char *_import_session_path(struct dt_import_session_t *self, gboole
   }
 
   gchar *new_path = dt_variables_expand(self->vp, pattern, FALSE);
+  g_free(pattern);
+
 #ifdef WIN32
   if(new_path && (strlen(new_path) > 1))
   {
     const char first = g_ascii_toupper(new_path[0]);
     if(first >= 'A' && first <= 'Z' && new_path[1] == ':') // path format is <drive letter>:\path\to\file
       new_path[0] = first;                                 // drive letter in uppercase looks nicer
   }
+  /* Remove trailing spaces from each element in the the path separated by directory separator.
+     So follow the usage in Windows. */
+  if(new_path && (strlen(new_path) >= 1))
```

Comment:
I would say this is suspicious. First it is possible to have a space in a path and second if we end-up with some wrong full path name here it means that we have missed some clean-up at some point before.

Reply:
I have tried it in Windows 11. If there it is a trailing space after any of the folder names (base directory or subdirectory in "copy & import") then the created folders will have the name without the trailing spaces. As a consequence Darktable is not able to import the selected pictures. ("Imported 0 image.")
In Ubuntu (Linux) the trailing spaces are absolutely OK in the folder names. So I added this modification only for Windows.

I chose this place for the cleaning up, because the user can give several folders in the path both for base and sub directory, and the trailing spaces must be removed also from the internal folders to make windows import the images. This is the place in the code when we are aware that the expanded variable is a path.

If you agree I can create a utility function from this code for expanding path variables, which expands a path and cleans it up, that is removes the trailing spaces for WIN32 etc.

Reply:
I would track down where the directory gets created and ensure that as on GNU/Linux we have spaces kept when creating the directories. Can you track this down and report where this is happening? (I mean where the spaces are ignored as you described). From there we can discuss a possible better solution. TIA.

Reply:
I try to find out what API is used for Windows.
A,
1. So far I am sure that Darktable on GNU/LINUX works properly. I have created different directory paths for the "copy&import" function. Folder names with starting and trailing spaces worked fine. The folders was created with the name containing the spaces, the images placed in the correct folders and displayed in Darktable. 
2. For GNU/LINUX only the trailing spaces after the extension caused problems.
B,

1. For Windows naming convention I have found the following:
"Do not end a file or directory name with a space or a period. Although the underlying file system may support such names, the Windows shell and user interface does not."  (https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file?redirectedfrom=MSDN#naming-conventions)

Reply:
Thanks, I think I understand this yet I'm pretty sure a directory with spaces like this is valid on Windows:

`c:\some dir\sub dir \ a file.txt`

Reply:
You may want to try this Python script:

```
import os

dir = os.path.join("sub dir ", " dir")
os.makedirs(dir)
```

On my side GNU/Linux this gives me:

```
$ ls -d sub*
'sub dir '

$ ls 'sub dir '
' dir'
```

Reply:
And my point is we want to fix that on darktable side properly by ensuring that we are using a proper set of calls keeping the spaces as-is.

Reply:
@TurboGit - you are right.
So our new goal in this issue, to find out which interface is called by Darktable and which should be called instead of.
I am trying to find it out, but I am new to windows, so it will take some time.

Reply:
@TurboGit - I have investigated the problem a bit more and found the following.
1, This part of Darktable code uses glib and gtk to access the filesystem. gtk in turn also uses glib. Glib gives us a common interface for any supported op. systems (Linux, Windows,...). If  Darktable runs on Windows, then for directory creation glib finally calls _wmkdir in Microsoft C Runtime Library. So far the directory path/name was not changed only UTF-8 text was converted into UTF-16 text.
2, _wmkdir makes some edits on the path, among others removes the trailing spaces from the name of the folder to be created. (I have tried out your python script on Win11. It created "sub dir" without the trailing space and the next line gave a FileNotFoundError.)
3, You can force Windows not doing edit on the path by prefixing the path with the \\?\ string. Even it can be done on Darktable interface.
![image](https://user-images.githubusercontent.com/14832992/210149785-db8710e3-3570-4923-95ec-698528965c78.png)
4, If you somehow creates folders with trailing spaces, you will have other problems in Darktable. Eg these folders will not appear under "folders" on "copy & import" neither on "add to library" window.

So the folder name with trailing space is a problem on windows only.
I see the following options:
1, Do not change anything - it is Windows - it behaves so.
2, Help the user by visualizing the trailing spaces better.
3, For Windows, and only for Windows remove silently the trailing spaces (except when the path starts with the magic \\?\ string.

And there is a separate problem: the trailing space after the filename which confuses the file extension. (This one I would remove in case of any op system.

Reply:
And there is one more possibility the 4th one:
For Windows, and only for Windows add the magic prefix \\\\?\ to the path if it is not there.
So the wanted folder names are created, and the import of images is successful. Although later on the trailing spaces will cause problems in Darktable, but the user receives, what he asked for.

Reply:
@erzsig : First, thanks for the details analysis. Now I'm thinking that either 3 or 4 are proper options and it will makes dt a bit more friendly to Windows user's messing the path. Last option (4) seems the easiest but probably not really good, a trailing space is certainly not wanted. So I'll probably go for 3 even if probably a bit more challenging and yes I think we should also remove filename trailing spaces.

If you are ok, please go with 3.

Reply:
@TurboGit: I agree with your choice, option 3 was my favorite. I have finished implementing the 3rd option (removing trailing spaces.) It was was very close to the original solution, I have only added minor changes. I have also rebased the code. Please review it again!

Reply:
Thanks! Will test now.